src/Controllers/MoviesController.cs

```diff
@@ -45,12 +47,18 @@ public IActionResult Create(MovieViewModel mvm)
         }
 
         [HttpGet("Index")]
-        public IActionResult Index(string movieGenre, string searchString, string sortOrder) 
+        public IActionResult Index(string movieGenre, string searchString, string sortOrder, int? pageNumber, int? pageSize) 
         {
-            var movies = UnitOfWork.Movies.GetAll().ToList();
             ViewData["TitleSortParam"] = String.IsNullOrEmpty(sortOrder) ? "title_desc" : "";
             ViewData["ReleaseDateSortParam"] = sortOrder == "Release Date" ? "date_desc" : "ReleaseDate";
             ViewData["GenreSortParam"] = sortOrder == "Genre" ? "genre_desc" : "Genre";
+            ViewData["CurrentSort"] = sortOrder;
+            ViewData["CurrentFilter"] = movieGenre;
+            if (searchString != null)
+                pageNumber = 1;
+            if (!pageSize.HasValue)
+                pageSize = 3;
```

Comment:
Define this value in a private constant var or in the Config file. Then, you can ask the same using the method params : i.e. `method(string a = "fede")` set `a` with "fede" if the method doesn't receive any value.